src/app/core/data/base/create-data.spec.ts

```diff
@@ -0,0 +1,112 @@
+    protected rdbService: RemoteDataBuildService,
+    protected objectCache: ObjectCacheService,
+    protected halService: HALEndpointService,
+    protected notificationsService: NotificationsService,
+  ) {
+    super(undefined, requestService, rdbService, objectCache, halService, notificationsService, undefined);
+  }
+
+  public getBrowseEndpoint(options: FindListOptions = {}, linkPath: string = this.linkPath): Observable<string> {
+    return observableOf(endpoint);
+  }
+}
+
+describe('CreateDataImpl', () => {
+  let service: TestService;
+  let requestService;
+  let halService;
+  let rdbService;
+  let objectCache;
+  let notificationsService;
+  let selfLink;
+  let linksToFollow;
+  let testScheduler;
+  let remoteDataMocks;
+
+  function initTestService(): TestService {
+    requestService = getMockRequestService();
+    halService = new HALEndpointServiceStub('url') as any;
+    rdbService = getMockRemoteDataBuildService();
+    objectCache = {
+
+      addPatch: () => {
+        /* empty */
+      },
+      getObjectBySelfLink: () => {
+        /* empty */
+      },
+      getByHref: () => {
+        /* empty */
+      },
+    } as any;
+    notificationsService = {} as NotificationsService;
+    selfLink = 'https://rest.api/endpoint/1698f1d3-be98-4c51-9fd8-6bfedcbd59b7';
+    linksToFollow = [
+      followLink('a'),
+      followLink('b'),
+    ];
+
+    testScheduler = new TestScheduler((actual, expected) => {
+      // asserting the two objects are equal
+      // e.g. using chai.
+      expect(actual).toEqual(expected);
+    });
+
+    const timeStamp = new Date().getTime();
+    const msToLive = 15 * 60 * 1000;
+    const payload = { foo: 'bar' };
+    const statusCodeSuccess = 200;
+    const statusCodeError = 404;
+    const errorMessage = 'not found';
+    remoteDataMocks = {
+      RequestPending: new RemoteData(undefined, msToLive, timeStamp, RequestEntryState.RequestPending, undefined, undefined, undefined),
+      ResponsePending: new RemoteData(undefined, msToLive, timeStamp, RequestEntryState.ResponsePending, undefined, undefined, undefined),
+      Success: new RemoteData(timeStamp, msToLive, timeStamp, RequestEntryState.Success, undefined, payload, statusCodeSuccess),
+      SuccessStale: new RemoteData(timeStamp, msToLive, timeStamp, RequestEntryState.SuccessStale, undefined, payload, statusCodeSuccess),
+      Error: new RemoteData(timeStamp, msToLive, timeStamp, RequestEntryState.Error, errorMessage, undefined, statusCodeError),
+      ErrorStale: new RemoteData(timeStamp, msToLive, timeStamp, RequestEntryState.ErrorStale, errorMessage, undefined, statusCodeError),
+    };
+
+    return new TestService(
+      requestService,
+      rdbService,
+      objectCache,
+      halService,
+      notificationsService,
+    );
+  }
+
+  beforeEach(() => {
+    service = initTestService();
+  });
+
+  // todo: add specs (there were no ceate specs in original DataService suite!)
```

Comment:
@tdonohue I'd be happy to, but we would need to update the duration a bit.

* I think doing this thoroughly would take another **2-4h**, and we're already pushing 32h (not counting Art's time)
* Along with the CREATE/PUT/... specs I'd also like to add more specs for the actual composition, like [these ones ](https://github.com/DSpace/dspace-angular/pull/1791/files#diff-36a570953cdd4d885f59d700f963962fc631613bfecea4979b2a501af973cfaaR27)
  During the refactor I'd noticed that it's quite easy to overlook small mistakes when writing the "patch-through" methods; having these basic test methods we can reuse for each concrete service would give some peace of mind.
  This would be another **1-2h**.

Reply:
@ybnd : Since it sounds like these additional tests will take less than a day (maybe only 1/2 a day), I'd like to go ahead with them in this PR, as I think it makes it easier to trust this refactor & provides more stability going forward. 

I'll bump the initial `30h` estimate to `38h`... which should at least give you 6 more hours to complete these tasks.  If the total still ends up over by a bit, that's OK. But, we obviously want to get as close to this estimate as possible. Thanks!

Reply:
Added specs as discussed.

Simplified some of the `*Data` interface signatures to make them easier to test and implement
* No optional methods
* Promoted one of the previously optional method as it seemed useful
* Refactored out the usages of the other ones